src/util/ConfigManager/ConfigOptionProxy.h

```diff
@@ -86,8 +88,10 @@ class ConstConfigOptionProxy
 // value, that the configuration option holds.
 template <typename T>
 class ConfigOptionProxy
-    : public detail::ConfigOptionProxyImplementation<T, ConfigOption> {
-  using Base = detail::ConfigOptionProxyImplementation<T, ConfigOption>;
+    : public ConfigOptionImpl::ConfigOptionProxyImplementation<T,
+                                                               ConfigOption> {
+  using Base =
+      ConfigOptionImpl::ConfigOptionProxyImplementation<T, ConfigOption>;
```

Comment:
I think you can also move the `Proxy` into the namespace
(again: the same namepace for everything config related)
then you also have the shorter namespaces here.
You can still use a using alias outside again if you want.